### DISCUSSION TITLE
How to change the tsconfig.json file used for Storybook?

### DISCUSSION THREAD

**Discussion by henrywoody** [storybookjs/storybook#26534] [Help]
### Summary

I have a project using React and TypeScript with Storybook with a monorepo setup. To get code references to work better (opt+click in VSCode takes you to the source file rather than the output `.d.ts` file in the `dist` directory), I've set the `baseUrl` property of my `tsconfig.json` file to be the project root. However this causes issues when building the Storybook app (internal `.d.ts` files are not included in the builds of other packages and types within those packages that rely on those declaration files cause errors). I've found that removing the `baseUrl` property fixes the issue.

For this reason, I'd like to use a separate `tsconfig.json` file for Storybook that is different from the default one I use for the project (this is a common pattern for example with different environments or for tests).

**How can I set Storybook to use my other tsconfig file (e.g. `tsconfig.storybook.json`)?**

**Comment by valentinpalkovic** [comment]
Could you provide a minimal reproduction so that it is easier to play around?

**Reply by henrywoody** [reply]
Hi Valentin, I will put one together in the next few days, but I don't think the question requires a repro because it's just about changing the tsconfig file used by Storybook/Webpack.

For example, I use typedoc to generate regular docs and that has a very straightforward way to change the tsconfig file used (https://typedoc.org/options/configuration/#tsconfig): just do `typedoc --tsconfig path/to/tsconfig.json`.

**Reply by valentinpalkovic** [reply]
Hi @henrywoody 

I think a reproduction would still be helpful
because different parts of Storybook might require tsconfig configurations and before actually providing one solution for your problem, I would like to understand deeply the issue before implementing your proposed solution or suggesting another one.
